# What genes are differentially expressed in colorectal cancer?

**Names**:

1.  Joshua Gilo
2.  Daphne Janelyn Go
3.  Kyle Carlo Lasala

Objective: To identify differentially expressed genes in colorectal cancer using publicly available matched normal-cancer expression data.

```{r echo=TRUE, results='hide', message=FALSE, warning=FALSE, eval=FALSE}
library(TCGAbiolinks)
library(SummarizedExperiment)
library (readr)
library (dplyr)
library(tidyr)
library(ggplot2)
library(edgeR)
library(DESeq2)
```

### Downloading Gene Expression Data from TCGA

Using TCGAbiolinks, RNA-seq gene expression quantification data (HTSeq raw counts) for 20 tumor and 20 normal samples from the TCGA Colorectal Adenocarcinoma (TCGA-COAD) project were downloaded.

```{r echo=TRUE, results='hide', message=FALSE, warning=FALSE, eval=FALSE}
query_Target <- GDCquery(project = "TCGA-COAD",
                         data.category = "Transcriptome Profiling",
                         experimental.strategy = "RNA-Seq",
                         data.type = "Gene Expression Quantification")

```

```{r echo=TRUE, results='hide', message=FALSE, warning=FALSE, eval=FALSE}
samplesDown_Target <- getResults(query_Target, cols = c("cases"))
```

```{r echo=TRUE, results='hide', message=FALSE, warning=FALSE, eval=FALSE}
data_Target_normal <- TCGAquery_SampleTypes(barcode = samplesDown_Target,
                                         typesample = "NT")
```

```{r}
data_Target_tumor <- TCGAquery_SampleTypes(barcode = samplesDown_Target,
                                         typesample = "TP")
```

```{r echo=TRUE, results='hide', message=FALSE, warning=FALSE, eval=FALSE}
data_short_Target_normal <- data_Target_normal[1:20]
data_short_Target_tumor <- data_Target_tumor[1:20]

```

```{r echo=TRUE, results='hide', message=FALSE, warning=FALSE, eval=FALSE}
queryDown_Target_normal <- GDCquery(project = "TCGA-COAD",
                             data.category = "Transcriptome Profiling",
                             experimental.strategy = "RNA-Seq",
                             data.type = "Gene Expression Quantification",
                            barcode = data_short_Target_normal)
 

```

```{r echo=TRUE, results='hide', message=FALSE, warning=FALSE, eval=FALSE}
GDCdownload(queryDown_Target_normal, directory = "./normal")
```

```{r echo=TRUE, results='hide', message=FALSE, warning=FALSE, eval=FALSE}
queryDown_Target_tumor <- GDCquery(project = "TCGA-COAD",
                             data.category = "Transcriptome Profiling",
                             experimental.strategy = "RNA-Seq",
                             data.type = "Gene Expression Quantification",
                            barcode = data_short_Target_tumor)
 

```

```{r echo=TRUE, results='hide', message=FALSE, warning=FALSE, eval=FALSE}
GDCdownload(queryDown_Target_tumor, directory = "./tumor")
```

### Preprocessing TSV Files from TCGA

All downloaded TSV files were merged into a single data frame, with each sample annotated according to its group (Tumor or Control). The resulting table, containing genes as rows and samples as columns, is the required input format for DESeq2.

```{r echo=TRUE, results='hide', message=FALSE, warning=FALSE, eval=FALSE}

file_dir_norm <- "./normal/TCGA-COAD/Transcriptome_Profiling/Gene_Expression_Quantification"
file_dir_tumor <- "./tumor/TCGA-COAD/Transcriptome_Profiling/Gene_Expression_Quantification"


```

```{r echo=TRUE, results='hide', message=FALSE, warning=FALSE, eval=FALSE}

sample_files_norm <- list.files(file_dir_norm,
                        pattern = "\\.tsv$",
                        recursive = TRUE,
                        full.names = TRUE)

```

```{r echo=TRUE, results='hide', message=FALSE, warning=FALSE, eval=FALSE}

sample_files_tumor <- list.files(file_dir_tumor,
                        pattern = "\\.tsv$",
                        recursive = TRUE,
                        full.names = TRUE)

```

```{r echo=TRUE, results='hide', message=FALSE, warning=FALSE, eval=FALSE}
sample_names_norm <- data_short_Target_normal
sample_names_tumor <- data_short_Target_tumor

```

```{r}
sample_files <- c(sample_files_norm, sample_files_tumor)
sample_names <- c(sample_names_norm, sample_names_tumor)
labels <- c(
  rep("Normal", length(sample_files_norm)),
  rep("Tumor",  length(sample_files_tumor))
)
```

```{r echo=TRUE, results='hide', message=FALSE, warning=FALSE, eval=FALSE}
process_tsv <- function(file, sample_name, label) {
  read_tsv(file, col_types = cols(), comment = "#") %>%
    select(gene_id, gene_name, unstranded) %>%
    filter(!is.na(gene_name)) %>%
    mutate(
      unstranded = as.numeric(unstranded),
      sample = sample_name,
      group = label
    )
}
```

```{r echo=TRUE, results='hide', message=FALSE, warning=FALSE, eval=FALSE}
sample_data <- Map(process_tsv, sample_files, sample_names, labels)

```

```{r echo=TRUE, results='hide', message=FALSE, warning=FALSE, eval=FALSE}
long_df <- bind_rows(sample_data)
long_df
```

```{r echo=TRUE, results='hide', message=FALSE, warning=FALSE, eval=FALSE}
gene_map <- long_df %>%
  select(gene_id, gene_name) %>%
  distinct()
```

```{r echo=TRUE, results='hide', message=FALSE, warning=FALSE, eval=FALSE}
sample_df_T <- long_df %>%
  pivot_wider(
    id_cols = c(gene_id, gene_name), 
    names_from = sample, 
    values_from = unstranded
  )

sample_df_T

```

### PCA plot

We performed a Principal Component Analysis (PCA) to assess whether the Tumor and Normal samples cluster differently. Prior to PCA, the raw read counts were normalized using counts per million (CPM), which adjusts for sequencing depth. The CPM for each gene is calculated as:

$$
\text{CPM} = \frac{\text{raw counts for a gene}}{\text{total counts in the sample}} \times 10^6
$$\

```{r echo=TRUE, results='hide', message=FALSE, warning=FALSE, eval=FALSE}
selected_data <- sample_df_T[, 2:ncol(sample_df_T)]
counts <- data.frame(lapply(selected_data, as.numeric), check.names = FALSE)
counts[is.na(counts)] <- 0
counts <- as.matrix(counts)

depth<- colSums(counts)
keep<- is.finite(depth) & depth > 0
counts<- counts[, keep, drop = FALSE]
depth <- depth[keep]

cpm <- sweep(counts, 2, depth, "/") * 1e6
log_cpm <- log2(cpm + 1)

X <- t(log_cpm)
pca_result <- prcomp(X, center = TRUE, scale. = FALSE)

```

```{r echo=TRUE, results='hide', message=FALSE, warning=FALSE, eval=FALSE}

variance_explained <- (pca_result$sdev^2) / sum(pca_result$sdev^2) * 100

pc1_label <- paste0("PC1 (", round(variance_explained[1], 2), "%)")
pc2_label <- paste0("PC2 (", round(variance_explained[2], 2), "%)")

pca_data <- as.data.frame(pca_result$x)
pca_data$sample <- rownames(X)
pca_data$type <- factor(c(rep("Normal",20),rep("Tumor",20)))

library(ggrepel)

p <- ggplot(pca_data, aes(x = PC1, y = PC2, color = type, label = sample)) +
  geom_point(size = 4, alpha = 0.8) +
#  geom_text_repel(size = 3, max.overlaps = 20) +
  scale_color_manual(values = c("#1C1AAF", "#D4070F")) +
  labs(x = pc1_label, y = pc2_label) +
  theme_minimal()

print(p)
                        
ggsave(filename = "PCA.jpg", plot = p, width = 6, height = 6, dpi = 300)
```

**Analysis**

The PCA plot shows a clear separation between tumor (red) and normal (blue) samples along the first principal component (PC1).

### DESeq2

```{r echo=TRUE, results='hide', message=FALSE, warning=FALSE, eval=FALSE}
groups <- factor(c(rep("Normal",20),rep("Tumor",20)))
sample_df_de <- as.data.frame(sample_df_T)
sample_df_de <- sample_df_de[ , -2]
rownames(sample_df_de) <- sample_df_de[[1]]
sample_df_de[[1]] <- NULL
min_read <- 1
sample_df_de <- sample_df_de[apply(sample_df_de,1,function(x){max(x)}) > min_read,]
names(groups) <- colnames(sample_df_de) 
sampleInfo <- data.frame(groups,row.names=colnames(sample_df_de))
sampleInfo


```

```{r echo=TRUE, results='hide', message=FALSE, warning=FALSE, eval=FALSE}
dds <- DESeqDataSetFromMatrix(countData = sample_df_de, colData = sampleInfo, design = ~ groups)
```

```{r echo=TRUE, results='hide', message=FALSE, warning=FALSE, eval=FALSE}
dds$groups = relevel(dds$groups,ref="Normal")
dds <- DESeq(dds)
res <- results(dds)
res_df <- as.data.frame(res)
res_df

```

### Why are we using p-adj instead of pvalue?

The adjusted p-value (padj) is the p-value corrected for multiple testing using the Benjamini–Hochberg procedure, which controls the false discovery rate (FDR).

We additionally filter out genes with a log₂ fold change of zero, as this indicates no difference in expression (neither upregulation nor downregulation) between the tumor and normal groups.

```{r echo=TRUE, results='hide', message=FALSE, warning=FALSE, eval=FALSE}
resSig <- res[(!is.na(res$padj) & (res$padj < 0.050000) &(abs(res$log2FoldChange)> 0.000000)), ]

resSig_df <- as.data.frame(resSig)
resSig_df
```

```{r}
resSig_df$gene_id <- rownames(resSig_df)
rownames(resSig_df) <- NULL

```

```{r}
resSig_df <- resSig_df %>%
  left_join(gene_map, by = "gene_id")
resSig_df

```

### Most Significantly Differentially expressed genes in tumor samples in terms of padj

```{r}

ggplot(resSig_df,
  aes(x = log2FoldChange,
      y = -log10(padj),
      color = factor(
      ifelse(padj < 0.05 & log2FoldChange > 1, "Up",
      ifelse(padj < 0.05 & log2FoldChange < -1, "Down", "NS")),
      levels = c("Up", "Down", "NS")))) +
  geom_point() +
  scale_color_manual(name = "Expression",
                     values = c("Up" = "red", "Down" = "blue", "NS" = "grey")) +
  theme_minimal()

```

This volcano plot shows that many genes are significantly upregulated (red) or downregulated (blue) between the two groups (Tumor and Normal), indicating substantial differences in gene expression profiles. Genes in grey are not significantly different.

```{r}
resSig_df_padj <- resSig_df[order(resSig_df$padj), c("gene_id", "gene_name", "padj")]
head(resSig_df_padj, 10)
```

**Analysis**

insert analysis for top significantly expressed (Do top 3 here)

### Upregulated and Downregulated Genes

In a given comparison, a positive fold change denotes higher expression, whereas a negative value indicates lower expression. Fold change is typically expressed on a log₂ scale. For instance, a log2fold change of 1.5 in a Tumor vs Control comparison means the gene’s expression in Tumor Samples is approximately 2.82 times higher than in Control ($2^{1.5} \approx 2.82$).

#### Top significantly upregulated genes in tumor samples in terms of log2FoldChange

```{r}
resSig_df_upreg <- resSig_df[order(-resSig_df$log2FoldChange), c("gene_id", "gene_name", "log2FoldChange", "padj")]
resSig_df_upreg

```

**Analysis**

insert analysis for upregulated (Do top 5 here)

#### Top significantly downregulated genes in tumor samples in terms of log2FoldChange

```{r}
resSig_df_downreg <- resSig_df[order(resSig_df$log2FoldChange), c("gene_id", "gene_name", "log2FoldChange","padj")]
resSig_df_downreg
```

**Analysis**

insert analysis for downregulated (Do top 3 here)
